# HG changeset patch
# User Ernst Reissner <rei3ner@arcor.de>
# Date 1505392131 -7200
#      Thu Sep 14 14:28:51 2017 +0200
# Node ID 1d5c85655a8ff433a8498299b626dfcd4006186a
# Parent  8e8dbabd6e848e7343b0c315203e5f4f332535aa
doc: Review content of "scripts" folder for m-files (bug #51996).

* doc/interpreter/func.txi: Add entries: +containers, gui, java, and ode.

* etc/HACKING: Add entry "+containers".

Pushed by Kai T. Ohlhus <k.ohlhus@gmail.com>.

diff --git a/doc/interpreter/func.txi b/doc/interpreter/func.txi
--- a/doc/interpreter/func.txi
+++ b/doc/interpreter/func.txi
@@ -1912,51 +1912,63 @@ Many of Octave's standard functions are 
 They are loosely organized by topic, in subdirectories of
 @file{@var{octave-home}/lib/octave/@var{version}/m}, to make it easier
 to find them.
 
 The following is a list of all the function file subdirectories, and the
 types of functions you will find there.
 
 @table @file
+@item @@ftp
+Class functions for the FTP object.
+
+@item +containers
+Package for the containers classes.
+
 @item audio
 Functions for playing and recording sounds.
 
 @item deprecated
 Out-of-date functions which will eventually be removed from Octave.
 
 @item elfun
 Elementary functions, principally trigonometric.
 
-@item @@ftp
-Class functions for the FTP object.
-
 @item general
 Miscellaneous matrix manipulations, like @code{flipud}, @code{rot90},
 and @code{triu}, as well as other basic functions, like
 @code{ismatrix}, @code{narginchk}, etc.
 
 @item geometry
 Functions related to Delaunay triangulation.
 
+@item gui
+Functions for GUI elements like dialog, message box, etc.
+
 @item help
 Functions for Octave's built-in help system.
 
 @item image
 Image processing tools.  These functions require the X Window System.
 
 @item io
 Input-output functions.
 
+@item java
+Functions related to the Java integration.
+
 @item linear-algebra
 Functions for linear algebra.
 
 @item miscellaneous
 Functions that don't really belong anywhere else.
 
+@item ode
+Functions to solve ordinary differential equations (ODEs).
+
 @item optimization
 Functions related to minimization, optimization, and root finding.
 
 @item path
 Functions to manage the directory path Octave uses to find functions.
 
 @item pkg
 Package manager for installing external packages of functions in Octave.
diff --git a/etc/HACKING b/etc/HACKING
--- a/etc/HACKING
+++ b/etc/HACKING
@@ -262,16 +262,18 @@ An overview of the directory layout of O
 
   m4            -- m4 scripts used by configure during the build process
 
 
   scripts       -- functions written in the Octave language
 
     @ftp             ftp object class
 
+    +containers      container classes (Map)
+
     audio            play and record sound files (system dependent)
 
     deprecated       older deprecated functions
 
     elfun            elementary mathematical functions
 
     general          utility functions
 
